packages/n3h-mock/lib/index.js

```diff
@@ -163,51 +183,195 @@ class N3hMock extends AsyncClass {
           return
-        case 'handleGetDhtResult':
-          // erm... since we're fully connected,
-          // just redirect this back to itself for now...
-          opt.data.method = 'getDhtResult'
+        case 'handleFetchEntryResult':
+          // Note: opt.data is a FetchEntryResultData
+          // if this message is a response from our own request, do a publish
+          if (opt.data._id in this._requestBook) {
+            delete this._requestBook[opt.data._id]
+            this._getMemRef(opt.data.dnaAddress).mem.insert({
+              type: 'dhtEntry',
+              providerAgentId: opt.data.providerAgentId,
+              entryAddress: opt.data.address,
+              content: opt.data.content
+            })
+            return
+          }
+          // otherwise since we're fully connected, just redirect this back to itself for now...
+          opt.data.method = 'fetchEntryResult'
           this._ipc.send('json', opt.data)
           return
-        case 'getDhtMeta':
+        case 'fetchMeta':
           // erm... since we're fully connected,
           // just redirect this back to itself for now...
-          opt.data.method = 'handleGetDhtMeta'
+          opt.data.method = 'handleFetchMeta'
           this._ipc.send('json', opt.data)
           return
-        case 'handleGetDhtMetaResult':
-          // erm... since we're fully connected,
-          // just redirect this back to itself for now...
-          opt.data.method = 'getDhtMetaResult'
+        case 'handleFetchMetaResult':
+          // Note: opt.data is a FetchMetaResultData
+          // if its from our own request do a publish
+          if (opt.data._id in this._requestBook) {
+            delete this._requestBook[opt.data._id]
+            this._getMemRef(opt.data.dnaAddress).mem.insert({
+              type: 'dhtMeta',
+              providerAgentId: opt.data.providerAgentId,
+              entryAddress: opt.data.entryAddress,
+              attribute: opt.data.attribute,
+              content: opt.data.content
+            })
+            return
+          }
+          // otherwise since we're fully connected, just redirect this back to itself for now...
+          opt.data.method = 'fetchMetaResult'
           this._ipc.send('json', opt.data)
           return
+
+        case 'handleGetPublishingEntryListResult':
+          // Note: opt.data is EntryListData
+          // Mark my request as resolved and get bucketId from request
+          bucketId = this._checkRequest(opt.data._id)
+          if (bucketId === '') {
+            return
+          }
+          // get already known publishing list
+          let knownPublishingList = {}
+          if(bucketId in this._publishedEntryBook) {
+            knownPublishingList = this._publishedEntryBook[bucketId]
+          }
+          // Update my book-keeping on what this agent has.
+          // and do a getEntry for every new entry
+          for (let entryAddress in opt.data.entryAddressList) {
```

Comment:
is this doing what you want? if `opt.data.entryAddressList` is an array, a `for in` loop will put the array index in `entryAddress` not the array value. Perhaps you meant a `for of` loop?

Reply:
nice catch! fixed